internal/pkg/warewulfd/parser.go

```diff
@@ -44,8 +47,17 @@ func parseReq(req *http.Request) (parserInfo, error) {
 		ret.efifile = path_parts[2]
 	}
 	ret.hwaddr = hwaddr
+    ipaddrtemp = ""
+    ipaddr = strings.Split(req.RemoteAddr, ":")
+    // IPv6 addresses hextets are split on ":"
+    if len(ipaddr) > 2 {
+       for i := 0; i < len(ipaddr)-1; i++ {
+            ipaddrtemp += ipaddr[i] + ":"
+       }
+       ret.ipaddr6 = strings.TrimSuffix(ipaddrtemp, ":")
+    }
 	ret.ipaddr = strings.Split(req.RemoteAddr, ":")[0]
-	ret.remoteport, _ = strconv.Atoi(strings.Split(req.RemoteAddr, ":")[1])
+	ret.remoteport, _ = strconv.Atoi(strings.Split(req.RemoteAddr, ":")[len(ipaddr)-1])
```

Comment:
I think it's awkward to use `ipaddr` as a general-purpose indicator for the length of `strings.Split(req.RemoteAddr, ":")`. `strings.Split(req.RemoteAddr, ":")` could just be replaced here with `ipaddr`; but I think `parts` from my example above is clearer:

```
parts := string.Split(req.RemoteAddr, ":")
if len(parts) > 2 {
    ret.ipaddr6 = strings.Join(parts[:len(parts)-1], ":")
} else {
    ret.ipaddr = parts[0]
}
ret.remoteport, _ = strconv.Atoi(parts[len(parts)-1])
```